src/main/java/com/blamejared/crafttweaker/impl/recipes/Replacer.java

```diff
@@ -0,0 +1,389 @@
+     * fixed according to {@link NameUtils#fixing(String)}.</p>
+     *
+     * @param oldName The {@link ResourceLocation} of the name of the recipe that should be renamed.
+     * @param newName The new name of the recipe.
+     * @return This replacer for chaining.
+     *
+     * @docParam oldName <resource:minecraft:birch_sign>
+     * @docParam newName "damn_hard_birch_sign"
+     */
+    @ZenCodeType.Method
+    public Replacer explicitlyRename(final ResourceLocation oldName, final String newName) {
+        final String actualNewName = this.fix(newName, oldName);
+        if (this.userRenames.containsKey(oldName) && !this.userRenames.get(oldName).equals(actualNewName)) {
+            CraftTweakerAPI.logError(
+                    "The same old name '%s' has been specified twice for renaming with two different strings '%s' and '%s': only the former will apply",
+                    oldName,
+                    this.userRenames.get(oldName),
+                    newName
+            );
+            return this;
+        }
+        
+        this.userRenames.put(oldName, actualNewName);
+        return this;
+    }
+    
+    /**
+     * Specifies the {@link BiFunction} that will be used for renaming all recipes.
+     *
+     * <p>The first argument to the function is the {@link ResourceLocation} that uniquely identifies its name, whereas
+     * the second represents the default name for the recipe according to the default replacement rules. The return
+     * value of the function will then represent the new name of the recipe.</p>
+     *
+     * @param function The renaming function.
+     * @return This replacer for chaining
+     *
+     * @docParam function myFunction
+     */
+    @ZenCodeType.Method
+    public Replacer useForRenaming(final BiFunction<ResourceLocation, String, String> function) {
+        if (this.userRenamingFunction != null) {
+            final CodePosition position = PositionUtil.getZCScriptPositionFromStackTrace();
+            CraftTweakerAPI.logWarning(
+                    "%sA renaming function has already been specified for this replacer: the old one will be replaced",
+                    position == CodePosition.UNKNOWN? "" : position + ": "
+            );
+        }
+        
+        this.userRenamingFunction = function;
+        return this;
+    }
+    
+    /**
+     * Executes all replacements that have been queued on this replacer, if any.
+     */
+    @ZenCodeType.Method
+    public void execute() {
+        if (this.rules.isEmpty()) return;
+        CraftTweakerAPI.apply(
+                new ReplacerAction(
+                        this.targetedManagers,
+                        this.targetedRecipes,
+                        Collections.unmodifiableList(this.rules),
+                        this.targetedManagers.stream().flatMap(manager -> DEFAULT_EXCLUSIONS.computeIfAbsent(manager, this::gatherDefaultExclusions).stream()).collect(Collectors.toList()),
```

Comment:
Well, it is a bit of an edge-case, but, when two recipes with the same name in different managers exist:
1) Are both excluded even if only one manager was told to exclude the recipe
2) If both managers said to exclude it, this list would contain both names, so the user would see the recipe excluded two times

Just a heads-up, but can stay as-is